src/gui/activitylistmodel.cpp

```diff
@@ -0,0 +1,231 @@
+{
+    if( _activityLists.count() == 0 ) return true;
+
+    QMap<AccountState*, ActivityList>::const_iterator i = _activityLists.begin();
+    while (i != _activityLists.end()) {
+        AccountState *ast = i.key();
+        if( ast && ast->isConnected() ) {
+            ActivityList activities = i.value();
+            if( activities.count() == 0 &&
+                    ! _currentlyFetching.contains(ast) ) {
+                return true;
+            }
+        }
+        ++i;
+    }
+
+    return false;
+}
+
+void ActivityListModel::startFetchJob(AccountState* s)
+{
+    if( !s->isConnected() ) {
+        return;
+    }
+    JsonApiJob *job = new JsonApiJob(s->account(), QLatin1String("ocs/v1.php/cloud/activity"), this);
+    QObject::connect(job, SIGNAL(jsonReceived(QVariantMap, int)),
+                     this, SLOT(slotActivitiesReceived(QVariantMap, int)));
+    job->setProperty("AccountStatePtr", QVariant::fromValue<AccountState*>(s));
+
+    QList< QPair<QString,QString> > params;
+    params.append(qMakePair(QString::fromLatin1("page"),     QString::fromLatin1("0")));
+    params.append(qMakePair(QString::fromLatin1("pagesize"), QString::fromLatin1("100")));
+    job->addQueryParams(params);
+
+    _currentlyFetching.insert(s);
+    qDebug() << Q_FUNC_INFO << "Start fetching activities for " << s->account()->displayName();
+    job->start();
+}
+
+void ActivityListModel::slotActivitiesReceived(const QVariantMap& json, int statusCode)
+{
+    auto activities = json.value("ocs").toMap().value("data").toList();
+
+    ActivityList list;
+    AccountState* ast = qvariant_cast<AccountState*>(sender()->property("AccountStatePtr"));
+    _currentlyFetching.remove(ast);
+
+    foreach( auto activ, activities ) {
+        auto json = activ.toMap();
+
+        Activity a;
+        a._type = Activity::ActivityType;
+        a._accName  = ast->account()->displayName();
+        a._id       = json.value("id").toLongLong();
+        a._subject  = json.value("subject").toString();
+        a._message  = json.value("message").toString();
+        a._file     = json.value("file").toString();
+        a._link     = json.value("link").toUrl();
+        a._dateTime = json.value("date").toDateTime();
+        a._dateTime.setTimeSpec(Qt::UTC);
+        list.append(a);
+    }
+
+    _activityLists[ast] = list;
+
+    emit activityJobStatusCode(ast, statusCode);
+
+    combineActivityLists();
+}
+
+
+void ActivityListModel::combineActivityLists()
+{
+    ActivityList resultList;
+
+    foreach( ActivityList list, _activityLists.values() ) {
+        resultList.append(list);
+    }
+
+    std::sort( resultList.begin(), resultList.end() );
+
+    beginRemoveRows(QModelIndex(), 0, _finalList.count() );
+    _finalList.clear();
+    endRemoveRows();
+
+    beginInsertRows(QModelIndex(), 0, resultList.count());
+    _finalList = resultList;
+    endInsertRows();
+}
+
+void ActivityListModel::fetchMore(const QModelIndex &)
```

Comment:
Normaly, fetchMore is too be used when there are more items to fetch that are hidden.
So that can be used in order to fetch older items form example. 
But here it's used to fetch other accounts despite it's not the next item.
fetchMore will be called when the scrollbar reach the end. 
I don't think it's a good use of the fetchMore capabilities